pkg/compiler/backend/common.go

```diff
@@ -41,3 +48,45 @@ func NewBackend(s string) compiler.CompilerBackend {
 	}
 	return compilerBackend
 }
+
+func RunCommand(cmd *exec.Cmd) (string, error) {
```

Comment:
I don't think this function should run the command, but rather just set `cmd` in a way that is consumable from the backend.

If you don't plan to add unit tests, this can be a private function instead

Reply:
In the real, it runs the command. It uses `os/exec` but to handle stdout/stderr it's needed to split the execution into two calls: cmd.Start() and cmd.Wait() for trigger end flush. So, it runs the command. I don't understand your note.
About the test unit, I'm not sure. it's like creating a test unit for `os/exec` package. I will rename the function as private then.

Reply:
I see that you are running the command here, indeed what I mean in pseudocode is:

```golang
function mutateCmd(cmd) cmd

//...

func doWork() {

cmd = mutateCmd(createCmd())

}
// ... use cmd as usual
```

Reply:
About the first, to move out the cmd.Start and cmd.Wait doesn't sense to me if do you think about this.

Reply:
The unit test then would check what the mutation is expected to be - not testing `os/exec`. But I'm fine with the wrapper as it goes into a private function (and not willing to test it ;) )